### PR TITLE
[browser][websocket] Fix for Close Description validation

### LINKED ISSUE #45531: [browser][websockets][tests] Tests for CloseDescription Max Length are failing
There seems to be a max length verification for the websocket close description that browser is not checking for.

The tests are expecting an exception but the browser websocket implementation does not throw an exception.

```
 fail: [FAIL] System.Net.WebSockets.Client.Tests.CloseTest.CloseAsync_CloseDescriptionIsMaxLengthPlusOne_ThrowsArgumentException(server: ws://corefx-net-http11.azurewebsites.net/WebSocket/EchoWebSocket.ashx)
  info: Assert.Throws() Failure
  info: Expected: typeof(System.ArgumentException)
  info: Actual:   (No exception was thrown)
  info:    at System.AssertExtensions.Throws[ArgumentException](String expectedParamName, Action action)
  info:    at System.Net.WebSockets.Client.Tests.CloseTest.CloseAsync_CloseDescriptionIsMaxLengthPlusOne_ThrowsArgumentException(Uri server)
  info: --- End of stack trace from previous location ---

```

See https://github.com/dotnet/runtime/pull/45470 for tests

### PR DESCRIPTION
**Pull request by kjpou1**
The description validation exception was not being percolated through to the unit test

The tests are expecting an exception.

```
 fail: [FAIL] System.Net.WebSockets.Client.Tests.CloseTest.CloseAsync_CloseDescriptionIsMaxLengthPlusOne_ThrowsArgumentException(server: ws://corefx-net-http11.azurewebsites.net/WebSocket/EchoWebSocket.ashx)
  info: Assert.Throws() Failure
  info: Expected: typeof(System.ArgumentException)
  info: Actual:   (No exception was thrown)
  info:    at System.AssertExtensions.Throws[ArgumentException](String expectedParamName, Action action)
  info:    at System.Net.WebSockets.Client.Tests.CloseTest.CloseAsync_CloseDescriptionIsMaxLengthPlusOne_ThrowsArgumentException(Uri server)
  info: --- End of stack trace from previous location ---

```

Running the outerloop tests locally passes correctly with this modification which follows the `ManagedWebSocket.cs` verification sequence.

See https://github.com/dotnet/runtime/pull/45470 for tests

Fix https://github.com/dotnet/runtime/issues/45531

### CODE REVIEW

File: src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/BrowserWebSocket.cs
Comment (stephentoub) [inline src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/BrowserWebSocket.cs:473]: Can this Invoke call throw?

Comment (kjpou1) [reply src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/BrowserWebSocket.cs:473]: Yes, that is what is being captured here: https://github.com/dotnet/runtime/issues/45538

Comment (stephentoub) [reply src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/BrowserWebSocket.cs:473]: If this Invoke call can throw, then you need a try/catch that will catch that exception and store it into a task to be returned.

Comment (kjpou1) [reply src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/BrowserWebSocket.cs:473]: perfect thanks

Comment (kjpou1) [reply src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/BrowserWebSocket.cs]: Addressed in recent commits

File: src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/BrowserWebSocket.cs
Comment (stephentoub) [inline src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/BrowserWebSocket.cs:468]: This no longer needs to be a separate method, since it's no longer awaiting anything.  At that point you can then just combine this try/catch with the one in the caller.

Comment (kjpou1) [reply src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/BrowserWebSocket.cs:468]: True but did not want to do that because the CloseAsyncCore is called from other places.

Comment (stephentoub) [reply src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/BrowserWebSocket.cs:468]: Ok

**Comment by radical** [comment]
Just trying to understand the situation with tests here.
- Do all the *enabled* tests pass for websockets?
   - If so, then are there some disabled tests that this would be fixing?
      - If so, then can we enable those tests with this PR?

I'm just wondering if there are any tests that we'll be able to see, going from FAIL to PASS.

In the PR that you linked, I see some tests being marked with `ActiveIssue` for 45531, which I guess will be removed after this is merged.

**Comment by kjpou1** [comment]
The ActiveIssue for 45531 will be removed

**Comment by kjpou1** [comment]
Can we get another review?

**Comment by kjpou1** [comment]
Build iOS x64 Release AllSubsets_Mono failing in all CI jobs https://github.com/dotnet/runtime/issues/45654

File: src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/BrowserWebSocket.cs
Comment (BrennanConroy) [inline src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/BrowserWebSocket.cs:484]: Unrelated to this PR. Are there plans to implement this method? If not, can it be marked as unsupported in the browser?

I know the browser doesn't provide an API for partial closes like this, but it would be very useful to fake it by clearing the websockets message handlers and calling `onclose` inline.

You can see what we did in SignalR to not have possible long delays between calling `close` and the `onclose` callback being invoked.
https://github.com/dotnet/aspnetcore/blob/0ace31c91138d258008f9547478d156a97f1a87e/src/SignalR/clients/ts/signalr/src/WebSocketTransport.ts#L144-L146
https://github.com/dotnet/aspnetcore/blob/0ace31c91138d258008f9547478d156a97f1a87e/src/SignalR/clients/ts/signalr/src/WebSocketTransport.ts#L156-L170

Comment (kjpou1) [reply src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/BrowserWebSocket.cs:484]: There is an issue set up here to look into this:  https://github.com/dotnet/runtime/issues/45468

Could you please comment on this issue so that we can follow up to get this working for you.

**Comment by kjpou1** [comment]
The failure for Build iOS x64 Release AllSubsets_Mono failing in all CI jobs is unrelated see issue: https://github.com/dotnet/runtime/issues/45468

**Review by stephentoub — APPROVED** [review]

**Pull request merged**
